pkg/proxy/apis/config/validation/validation.go

```diff
@@ -69,16 +76,23 @@ func Validate(config *kubeproxyconfig.KubeProxyConfiguration) field.ErrorList {
 	allErrs = append(allErrs, validateHostPort(config.MetricsBindAddress, newPath.Child("MetricsBindAddress"))...)
 
 	if config.ClusterCIDR != "" {
-		if config.FeatureGates[string(kubefeatures.IPv6DualStack)] {
-			cidrs := strings.Split(config.ClusterCIDR, ",")
-			for _, cidr := range cidrs {
-				if _, _, err := net.ParseCIDR(cidr); err != nil {
-					allErrs = append(allErrs, field.Invalid(newPath.Child("ClusterCIDR"), cidr, "must be a valid CIDR block (e.g. 10.100.0.0/16 or FD02::0:0:0/96)"))
-				}
+		cidrs := strings.Split(config.ClusterCIDR, ",")
+		dualStackEnabled := effectiveFeatures.Enabled(kubefeatures.IPv6DualStack)
+
+		switch {
+		// if DualStack and more than one cidr validate if there is at least one of each IP family
+		case dualStackEnabled && len(cidrs) > 1:
+			isDual, err := netutils.IsDualStackCIDRStrings(cidrs)
+			if err != nil || !isDual {
+				allErrs = append(allErrs, field.Invalid(newPath.Child("ClusterCIDR"), config.ClusterCIDR, "must be a valid DualStack CIDR (e.g. 10.100.0.0/16,fde4:8dba:82e1::/48)"))
```

Comment:
since we're giving an example here, we should note that the family of the first IP determines the default family for the cluster, so this example means the cluster defaults to IPv4

Reply:
if we got an error back (`err != nil`), wouldn't it be helpful to include?

Reply:
this is a good point, and maybe we should stress it more, the first IP subnet indicates the default family of the cluster.

Reply:
I was thinking more about this, and I think that kube-proxy will not have a notion of default IP family, apiserver does, but kube-proxy just install iptables/ipvs rules based on the services and endpoints.